integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/query/projection/LoadingProjectionIT.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.integrationtest.mapper.orm.query.projection;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+import org.assertj.core.api.Assertions;
+import org.hibernate.SessionFactory;
+import org.hibernate.search.mapper.orm.Search;
+import org.hibernate.search.mapper.orm.hibernate.FullTextQuery;
+import org.hibernate.search.mapper.orm.hibernate.FullTextSearchTarget;
+import org.hibernate.search.mapper.orm.hibernate.FullTextSession;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.GenericField;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.Indexed;
+import org.hibernate.search.util.impl.integrationtest.common.rule.BackendMock;
+import org.hibernate.search.util.impl.integrationtest.common.rule.StubSearchWorkBehavior;
+import org.hibernate.search.util.impl.integrationtest.common.stub.backend.StubBackendUtils;
+import org.hibernate.search.util.impl.integrationtest.orm.OrmSetupHelper;
+import org.hibernate.search.util.impl.integrationtest.orm.OrmUtils;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class LoadingProjectionIT {
+
+	private static final String TITLE_4_3_2_1 = "4 3 2 1";
+	private static final String AUTHOR_4_3_2_1 = "Paul Auster";
+
+	private static final String TITLE_CIDER_HOUSE = "The Cider House Rules";
+	private static final String AUTHOR_CIDER_HOUSE = "John Irving";
+
+	private static final String TITLE_AVENUE_OF_MYSTERIES = "Avenue of Mysteries";
+	private static final String AUTHOR_AVENUE_OF_MYSTERIES = "John Irving";
+
+	@Rule
+	public BackendMock backendMock = new BackendMock( "stubBackend" );
+
+	@Rule
+	public OrmSetupHelper ormSetupHelper = new OrmSetupHelper();
+
+	private SessionFactory sessionFactory;
+
+	@Before
+	public void setup() {
+		backendMock.expectAnySchema( Book.INDEX );
+
+		sessionFactory = ormSetupHelper.withBackendMock( backendMock )
+				.setup( Book.class );
+
+		backendMock.verifyExpectationsMet();
+	}
+
+	@Test
+	public void index() {
+		OrmUtils.withinTransaction( sessionFactory, session -> {
```

Comment:
The database is cleared between each test method, so if you want to use the persisted data, you have to persist it in the same test method as you use it. In this case, you need to persist it in `search`.

You will tell me "but the test pass!" or "you did the same in the smoke tests" and I will agree: the tests are buggy, and it's concerning.

I think it's because the calls to `session.get( Book.class, 1 )` return null, and similarly the projection returns null books because the database is empty.

An easy way to fix all this would be to move this indexing code to a private `initData` method and call that method in `setup`.

Reply:
Ah, I saw it at one point but forgot to change it. Done now.